### PR TITLE
fix c-warnings; fix CI

### PR DESCRIPTION
**Pull request by tamird**
@sodabrew this should fix travis

### CODE REVIEW

**Comment by tamird** [comment]
travis failure is our old friend `dlopen(e, 9): image not found - e1`

File: ext/mysql2/statement.h [hunk 1]
```diff
@@ -13,6 +13,6 @@ void init_mysql2_statement(void);
 void decr_mysql2_stmt(mysql_stmt_wrapper *stmt_wrapper);
 
 VALUE rb_mysql_stmt_new(VALUE rb_client, VALUE sql);
-void rb_raise_mysql2_stmt_error(mysql_stmt_wrapper *stmt_wrapper);
+void rb_raise_mysql2_stmt_error(mysql_stmt_wrapper *stmt_wrapper) __attribute__ ((noreturn));
```
Comment (sodabrew) [inline ext/mysql2/statement.h]: Should be wrapped in a "GCC-ism" macro?

Comment (tamird) [inline ext/mysql2/statement.h]: aren't unknown attributes ignored by compilers that don't support them? what did the build do on appveyor?

**Comment by sodabrew** [comment]
Both Travis and AppVeyor built master fine today. Where did you see these warnings for build failures?

**Comment by tamird** [comment]
The travis badge on the readme is red, and here's the build for current HEAD https://travis-ci.org/brianmario/mysql2/builds/79690310

**Comment by sodabrew** [comment]
Right - that's what I mean. The failures are due to specs failing in the already-known ways, not compile failures.

**Comment by tamird** [comment]
Looks like a compile error to me:
https://travis-ci.org/brianmario/mysql2/jobs/79690325
https://travis-ci.org/brianmario/mysql2/jobs/79690328

**Comment by sodabrew** [comment]
I need to enroll in school.
cantreadgood

File: ext/mysql2/statement.h (hunk 1 above)
Comment (sodabrew) [inline ext/mysql2/statement.h]: Unknown attributes are no longer ignored in GCC https://gcc.gnu.org/bugzilla/show_bug.cgi?id=55768  -- could be true for other compilers that don't ignore and don't know about this one. There's something similar for the UNUSED macro: https://github.com/brianmario/mysql2/blob/master/ext/mysql2/mysql2_ext.h#L31-L35

Comment (tamird) [inline ext/mysql2/statement.h]: done

**Pull request merged**